lib/src/main/java/com/auth0/android/lock/events/DatabaseSignUpEvent.java

```diff
@@ -66,19 +68,14 @@ public void setRootAttributes(@NonNull Map<String, Object> attributes) {
      * @param customFields user_metadata fields to set
      */
     public void setExtraFields(@NonNull Map<String, String> customFields) {
-        this.rootAttributes.put(KEY_USER_METADATA, customFields);
+        this.userMetadata.putAll(customFields);
     }
 
 
     @SuppressWarnings("ConstantConditions")
     @NonNull
     public SignUpRequest getSignUpRequest(@NonNull AuthenticationAPIClient apiClient, @NonNull String connection) {
-        SignUpRequest request;
-        if (getUsername() != null) {
-            request = apiClient.signUp(getEmail(), getPassword(), getUsername(), connection);
-        } else {
-            request = apiClient.signUp(getEmail(), getPassword(), connection);
-        }
+        SignUpRequest request = apiClient.signUp(getEmail(), getPassword(), getUsername(), connection, userMetadata);
```

Comment:
user metadata will be passed as an extra param of this method call, rather than later in the request parameters. The method accepts a nullable username, so we no longer need the if